### PR TITLE
EventsManager changed from HQL to lucene search usage

### PR DESCRIPTION
**Pull request by msladek**
And related refactoring in Calendar, CalendarUtils and CalendarService

### CODE REVIEW

File: src/main/java/com/celements/calendar/manager/EventsManager.java [hunk 1]
```diff
@@ -36,87 +35,79 @@
...
 
-  @Requirement
-  ICalendarService calService;
-
   //TODO we must change to 'default' serializer with the wikiname included
   @Requirement("local")
   private EntityReferenceSerializer<String> refLocalSerializer;
 
-  @Requirement("default") EntityReferenceSerializer<String> refDefaultSerializer;
+  @Requirement("default")
```
Comment (ebeutler) [inline src/main/java/com/celements/calendar/manager/EventsManager.java]: Please use getRefDefaultSerializer() from IWebUtilsService

File: src/main/java/com/celements/calendar/service/CalendarService.java [hunk 2]
```diff
@@ -34,23 +51,84 @@
...
+  	}
   }
 
+  @Deprecated
   public String getAllowedSpacesHQL(XWikiDocument calDoc) throws XWikiException {
-    return CalendarUtils.getInstance().getAllowedSpacesHQL(calDoc, getContext());
+    return getAllowedSpacesHQL(calDoc.getDocumentReference());
+  }
+
+  public String getAllowedSpacesHQL(DocumentReference calDocRef) throws XWikiException {
```
Comment (ebeutler) [inline src/main/java/com/celements/calendar/service/CalendarService.java]: Shouldn't this method also be deprecated?

**Comment by ebeutler** [comment]
Looks good, just two small things.

**Comment by msladek** [comment]
thanks, they are fixed now

**Comment by msladek** [comment]
changes still pending, please don't merge yet

**Comment by msladek** [comment]
Please re-review changes: EventsManager uses CalendarEngineHQL or CalendarEngineLucene, depending on settings in Calendar.

**Comment by ebeutler** [comment]
On small thing in the new changes.

**Comment by msladek** [comment]
please re-re-review: CalendarClasses refactoring and getMidnightDate() moved to Calendar

**Pull request merged**